### PR TITLE
jenkins: add backwards compatibility test

### PR DESCRIPTION
**Pull request by JessicaGreben**
What/Why:

Have Jenkins run a backwards compatibility test on the master branch to so that we can check we don't break it.

Please describe the tests:
 - Test 1: backward compatibility test where we upload a file on the most recent release tag, then download the same file on current master.

Please describe the performance impact:
This will make running tests slower, so we will only run this test on the master branch instead of on every branch.

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: Jenkinsfile.public [hunk 1]
```diff
@@ -90,6 +90,23 @@ pipeline {
                         sh 'make test-sim'
                     }
                 }
+
+                stage('Backwards Compatibility') {
+                    when {
```
Comment (egonelbre) [inline Jenkinsfile.public]: You can use:
```
when {
    branch 'master'
}
```

File: Jenkinsfile.public [hunk 2]
```diff
@@ -90,6 +90,23 @@ pipeline {
...
+                    expression {
+                        return env.BRANCH_NAME == 'master';
+                        }
```
Comment (egonelbre) [inline Jenkinsfile.public]: Indentation messed up.

File: Jenkinsfile.public (hunk 1 above)
Comment (JessicaGreben) [reply Jenkinsfile.public]: i tried that here https://github.com/storj/storj/pull/2774/commits/4cab79887a3da97ed08f2199b432d201c16cec36, but it didn't run the tests. https://build.ops.storj.io/blue/organizations/jenkins/storj/detail/PR-2774/3/pipeline

File: scripts/test-sim-backwards.sh [hunk 3]
```diff
@@ -38,6 +38,8 @@ storj-sim -x --host $STORJ_NETWORK_HOST4 network test bash "$SCRIPTDIR"/test-bac
 
 make -C "$BRANCH_DIR" install-sim
 
+sed -i -e 's#/release/#/branch/#g' $STORJ_NETWORK_DIR/satellite/0/config.yaml
```
Comment (egonelbre) [inline scripts/test-sim-backwards.sh]: This replacement looks weird. I'm wondering what lines is it replacing?

Comment (JessicaGreben) [reply scripts/test-sim-backwards.sh]: its replacing anywhere that have "/release/" in the config file

File: Jenkinsfile.public (hunk 1 above)
Comment (egonelbre) [reply Jenkinsfile.public]: Hmm, weird. I wonder if it's handling PR-s with different branch names.

File: scripts/test-sim-backwards.sh (hunk 3 above)
Comment (JessicaGreben) [reply scripts/test-sim-backwards.sh]:
```
$ cat /var/folders/s_/n4czty1x4tj297495m66bcsc0000gp/T/tmp.XXXXXXXXXX.NBgZoieO/satellite/0/config.yaml | grep "/release/"
console.static-dir: /private/var/folders/s_/n4czty1x4tj297495m66bcsc0000gp/T/tmp.XXXXXXXXXX.NBgZoieO/release/web/satellite
mail.template-path: /private/var/folders/s_/n4czty1x4tj297495m66bcsc0000gp/T/tmp.XXXXXXXXXX.NBgZoieO/release/web/satellite/static/emails
marketing.static-dir: /private/var/folders/s_/n4czty1x4tj297495m66bcsc0000gp/T/tmp.XXXXXXXXXX.NBgZoieO/release/web/marketing```

Comment (egonelbre) [reply scripts/test-sim-backwards.sh]: Let's add a comment that this renames the static dir paths.

File: Jenkinsfile.public (hunk 1 above)
Comment (JessicaGreben) [reply Jenkinsfile.public]: i don't have a preference to run it for master or all PRs.

Comment (egonelbre) [reply Jenkinsfile.public]: Let's run it for all, until it starts causing problems.

Comment (egonelbre) [reply Jenkinsfile.public]: One thing we could test is to run it after `make test-sim` in Integration. That would mean it'll use less concurrency and might be faster.

**Pull request merged**